### PR TITLE
[Requirements] Bump numpy for tensorflow2.14.0

### PR DESCRIPTION
**Pull request by tomerm-iguazio**
[ML-4855](https://jira.iguazeng.com/browse/ML-4855)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -3,10 +3,11 @@ v3io~=0.5.14
 # exclude pandas 1.5.0 due to https://github.com/pandas-dev/pandas/issues/48767
 # and 1.5.* due to https://github.com/pandas-dev/pandas/issues/49203
 pandas>=1, !=1.5.*, <3
-numpy>=1.16.5,<1.23
+# <=1.26.2 is just a safeguard - no tests performed with numpy higher than 1.26.2
+numpy<=1.26.2
```
Comment (gtopper) [inline requirements.txt]: Please don't limit the patch version though.
```suggestion
# upper limit is just a safeguard - tested with numpy 1.26.2
numpy<1.27
```

File: requirements.txt [hunk 2]
```diff
@@ -3,10 +3,11 @@ v3io~=0.5.14
...
 # <15 is just a safeguard - no tests performed with pyarrow higher than 14
 pyarrow>=1,<15
 v3io-frames~=0.10.3
 v3iofs~=0.1.17
 xxhash>=1
-nuclio-sdk>=0.5.3
\ No newline at end of file
+nuclio-sdk>=0.5.3
```
Comment (gtopper) [inline requirements.txt:13]: What's the change?

File: setup.py [hunk 3]
```diff
@@ -38,8 +38,8 @@ def load_deps(file_name):
 install_requires = load_deps("requirements.txt")
 tests_require = load_deps("dev-requirements.txt")
 extras_require = {
-    "s3": ["s3fs~=0.5"],
-    "az": ["adlfs~=0.5"],
+    "s3": ["s3fs==2023.9.2"],
+    "az": ["adlfs==2023.9.0"],
```
Comment (gtopper) [inline setup.py]: Pretty sure you have a separate PR for this.

Comment (gtopper) [reply setup.py]: #481

File: requirements.txt (hunk 2 above)
Comment (tomerm-iguazio) [reply requirements.txt:13]: Idk and I can not revert it, maybe it's just a GitHub mistake.

Install requirements works and install the required package, So I think we can ignore this bug.

File: setup.py (hunk 3 above)
Comment (tomerm-iguazio) [reply setup.py]: Your'e right. Reverted.

File: requirements.txt (hunk 2 above)
Comment (assaf758) [reply requirements.txt:13]: it's a formatting change - adding newline char at the end of file

**Review by gtopper — APPROVED** [review]

File: requirements.txt [hunk 4]
```diff
@@ -3,10 +3,12 @@ v3io~=0.5.14
 # exclude pandas 1.5.0 due to https://github.com/pandas-dev/pandas/issues/48767
 # and 1.5.* due to https://github.com/pandas-dev/pandas/issues/49203
 pandas>=1, !=1.5.*, <3
-numpy>=1.16.5,<1.23
+# numpy>=1.23 in order to support tensorflow 2.14.0 in mlrun.
```
Comment (gtopper) [inline requirements.txt]: Why do we need to raise the lower bound in storey to support something in mlrun? Let mlrun specify its own limits...

**Review by gtopper — CHANGES_REQUESTED** [review]

File: requirements.txt [hunk 5]
```diff
@@ -3,10 +3,12 @@ v3io~=0.5.14
...
+# upper limit is just a safeguard - tested with numpy 1.26.2
+numpy>=1.23.0,<1.27
```
Comment (gtopper) [inline requirements.txt]:
```suggestion
# upper limit is just a safeguard - tested with numpy 1.26.2
numpy>=1.16.5,<1.27
```

**Review by assaf758 — APPROVED** [review]

**Pull request merged**